src/OpenTelemetry.Instrumentation.Runtime/description.md

```diff
@@ -0,0 +1,198 @@
+# Runtime metrics description
+
+Metrics name are prefixed with the `process.runtime.dotnet.` namespace, following
+the general guidance for runtime metrics in the
+[specs](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/runtime-environment-metrics.md#runtime-environment-specific-metrics---processruntimeenvironment).
+Instrument Units [should follow](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/README.md#instrument-units)
+the Unified Code for Units of Measure.
+
+## GC related metrics
+
+The metrics in this section can be enabled by setting the
+`RuntimeMetricsOptions.IsGcEnabled` switch.
+
+| Name                                          | Description              | Units     | Instrument Type   | Value Type | Attribute Key(s) | Attribute Values |
+|-----------------------------------------------|--------------------------|-----------|-------------------|------------|------------------|------------------|
+| process.runtime.dotnet.**gc.totalmemorysize** | Total allocated bytes    | `By`      | ObservableGauge   | `Int64`    |                  |                  |
+| process.runtime.dotnet.**gc.count**           | Garbage Collection count | `{times}` | ObservableCounter | `Int64`    | gen              | gen0, gen1, gen2 |
+
+Question for .NET team: is GC.GetTotalMemory(false) always equal to the sum of
```

Comment:
@tarekgh can you give comment on this?

Reply:
My understanding is that they aren't the same. GC.GetGCMemoryInfo() is giving a snapshot of historical data that was computed when the most recent GC started. GC.GetTotalMemory(false) is giving the live value. Aside from timing difference there may also be a difference in the accuracy. The GC has a central pool of memory that it gives out to threads in chunks (8 or 16KB each I think?). Each chunk is called an allocation context inside the runtime. Each thread then sub-allocates individual .NET objects from its private allocation context. I wouldn't be surprised if GetTotalMemory approximates that the entire allocation_context as being allocated memory whereas GetGCMemoryInfo() might be accurately measuring how much of each allocation_context has been given out by the thread which owns it.

Reply:
I've removed the metrics for `GC.GetTotalMemory(false)` because it includes the fragmentation and it is not very useful to customers.

Its return value is the size of: live object (not necessarily hold by the user objects/threads, could be hold by the GC itself) + fragmentation.

Reply:
Removing `gc.totalmemorysize` because it is duplicate and can be calculated from `gc.heapsize`.